features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/des/des_alt.c

```diff
@@ -293,36 +296,52 @@ int mbedtls_des3_crypt_ecb( mbedtls_des3_context *ctx,
  * 3DES-CBC buffer encryption/decryption
  */
 int mbedtls_des3_crypt_cbc( mbedtls_des3_context *ctx,
-                     int mode,
-                     size_t length,
-                     unsigned char iv[8],
-                     const unsigned char *input,
-                     unsigned char *output )
+                            int mode,
+                            size_t length,
+                            unsigned char iv[8],
+                            const unsigned char *input,
+                            unsigned char *output )
 {
     return mbedtls_des_docrypt(ctx->keyopt, ctx->key, mode == MBEDTLS_DES_ENCRYPT, TDES_MODE_CBC, length, iv, input, output);
 }
 #endif /* MBEDTLS_CIPHER_MODE_CBC */
 
 
 
-static int mbedtls_des_docrypt(uint16_t keyopt, uint8_t key[3][MBEDTLS_DES_KEY_SIZE], int enc, uint32_t tdes_opmode, size_t length, 
-    unsigned char iv[8], const unsigned char *input, unsigned char *output)
+static int mbedtls_des_docrypt(uint16_t keyopt, uint8_t key[3][MBEDTLS_DES_KEY_SIZE], int enc, uint32_t tdes_opmode, size_t length,
+                               unsigned char iv[8], const unsigned char *input, unsigned char *output)
 {
     if (length % 8) {
         return MBEDTLS_ERR_DES_INVALID_INPUT_LENGTH;
     }
+
+    /* DES DMA buffer has the following requirements:
+     * (1) Word-aligned buffer base address
+     * (2) 8-byte aligned buffer size
+     * (3) Located in 0x20000000-0x2FFFFFFF region
+     */
+    if ((! crypto_dma_buff_compat(dmabuf_in, MAXSIZE_DMABUF, 8)) ||
+        (! crypto_dma_buff_compat(dmabuf_out, MAXSIZE_DMABUF, 8))) {
+        error("Buffer for DES alter. DMA requires to be word-aligned and located in 0x20000000-0x2FFFFFFF region.");
+    }
+    
+    /* TODO: Change busy-wait to other means to release CPU */
+    /* Acquire ownership of DES H/W */
+    while (! crypto_des_acquire());
+    
+    /* Init crypto module */
+    crypto_init();
     
     // NOTE: Don't call driver function TDES_Open in BSP because it doesn't support TDES_CTL[3KEYS] setting.
```

Comment:
Why not using `TDES_Open` for the first write to `CRPT->TDES_CTL` and doing the `3KEYS` write manually, afterwards?

Reply:
@hanno-arm I corrected the comment in https://github.com/ARMmbed/mbed-os/pull/4925/commits/e2452a51e9e23b97fb85ac7caf86f62ec380465b. The cause is to avoid `CRPT->TDES_CTL` overridden by internal multiple channel support in DES BSP driver.

Reply:
@ccli8 Could you elaborate? I think the currently calling `TDES_Open` would be functionally equivalent to what you're doing (passing Channel `0`).

Reply:
@hanno-arm I changed it by calling BSP driver in https://github.com/ARMmbed/mbed-os/pull/4925/commits/5f229a3d33096787ad8a312bb561159a432fb7c0.